src/platform/plugins/shared/controls/public/controls/esql_control/types.ts

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+import { PublishesESQLVariable } from '@kbn/esql-variables-types';
+import type { ESQLControlState as ControlState } from '@kbn/esql/public';
+import type { HasEditCapabilities, PublishesPanelTitle } from '@kbn/presentation-publishing';
+import type { DefaultControlState } from '../../../common';
+import type { DefaultControlApi } from '../types';
+
+export interface ESQLControlState extends DefaultControlState, Omit<ControlState, 'width'> {}
```

Comment:
Having 2 types named `ESQLControlState` is a little confusing. Why is this version of `ESQLControlState` needed? What problem is this type definition solving? Would it be possible to use a single type definition for `ESQLControlState`?

Reply:
Sure https://github.com/elastic/kibana/pull/202875/commits/70c54ba7acb0304ecea576f8a734f6ebbfa03297